### PR TITLE
【autograd】add max_p primitive operator for new autograd 

### PR DESCRIPTION
**Pull request by betterpig**
### PR types

New features
### PR changes

OPs
### Describe

add max_p primitive operator for new autograd.
use eq and select to help compute z_dot.
Given x_dot and y_dot:
- x > y  --> y != z, return x_dot
- x == y --> y = z, return y_dot
- x < y --> y = z, return y_dot

### CODE REVIEW

File: python/paddle/incubate/autograd/primrules.py [hunk 1]
```diff
@@ -737,6 +750,31 @@ def _compute_t1(x, y):
...
+    if y_dot is None:
+        return mul(x_dot, _balanced_eq(x, z, y))
+    elif x_dot is None:
+        return mul(y_dot, _balanced_eq(y, z, x))
+    else:
+        t1 = mul(x_dot, _balanced_eq(x, z, y))
+        t2 = mul(y_dot, _balanced_eq(y, z, x))
+        z_dot = add(t1, t2)
+        return z_dot
+
```
Comment (cxxly) [inline python/paddle/incubate/autograd/primrules.py:771]: 当 `x==y` 时，paddle原生OP将梯度回传给了`y`，不是平分给`x`,`y`。

```python
...
half = fill_const(value=0.5, shape=z.shape, dtype=z.dtype)

if y_dot is None:
    x_jac = _balanced_eq(x, z, y)
    x_jac = select(eq(x, y), sub(x_jac-half), x_jac)
    return mu(x_dot, x_jac)
elif x_dot is None:
     ...
     y_jac = select(eq(x, y), add(y_jac + half), y_jac)
...
```

Comment (betterpig) [reply python/paddle/incubate/autograd/primrules.py:771]: 已改成和paddl.maximum一致的梯度实现
```
if y_dot is None:
    return select(eq(y, z), z_zeros, x_dot)
elif x_dot is None:
    return select(eq(y, z), y_dot, z_zeros)
else:
    return select(eq(y, z), y_dot, x_dot)
```

**Review by cxxly — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**